tests/unittests/wrappers/test_multitask.py

```diff
@@ -0,0 +1,182 @@
+_classification_preds = torch.randint(high=2, size=(NUM_BATCHES, BATCH_SIZE))
+_classification_target = torch.randint(high=2, size=(NUM_BATCHES, BATCH_SIZE))
+_multitask_preds = {"Classification": _classification_preds, "Regression": _regression_preds}
+_multitask_targets = {"Classification": _classification_target, "Regression": _regression_target}
+
+
+def _dict_results_same_as_individual_results(classification_results, regression_results, multitask_results):
+    return (
+        multitask_results["Classification"] == classification_results
+        and multitask_results["Regression"] == regression_results
+    )
+
+
+def _multitask_same_as_individual_tasks(classification_metric, regression_metric, multitask_metrics):
+    """Update classification and regression metrics individually and together using a multitask wrapper.
+
+    Return True if the results are the same.
+    """
+    classification_metric.update(_classification_preds, _classification_target)
+    regression_metric.update(_regression_preds, _regression_target)
+    multitask_metrics.update(_multitask_preds, _multitask_targets)
+
+    classification_results = classification_metric.compute()
+    regression_results = regression_metric.compute()
+    multitask_results = multitask_metrics.compute()
+
+    return _dict_results_same_as_individual_results(classification_results, regression_results, multitask_results)
+
+
+def test_errors_on_wrong_input():
+    """Check that type errors are raised when inputs are of the wrong type."""
+    with pytest.raises(TypeError, match="Expected *"):
+        MultitaskWrapper(1)
+
+    with pytest.raises(TypeError, match="Expected *"):
+        MultitaskWrapper(None)
+
+    with pytest.raises(TypeError, match="Expected *"):
+        MultitaskWrapper({"Classification": 1})
+
+
+def test_error_on_wrong_keys():
+    """Check that ValueError is raised when the sets of keys of the task metrics, preds, and targets do not match."""
+    multitask_metrics = MultitaskWrapper(
+        {
+            "Classification": BinaryAccuracy(),
+            "Regression": MeanSquaredError(),
+        }
+    )
+
+    # Classification preds, but not regression preds
+    wrong_key_preds = {"Classification": _classification_preds}
+
+    # Classification targets, but not regression targets
+    wrong_key_targets = {"Classification": _classification_target}
+
+    # Classification metric, but not regression metric
+    wrong_key_multitask_metrics = MultitaskWrapper(
+        {
+            "Classification": BinaryAccuracy(),
+        }
+    )
+
+    with pytest.raises(ValueError, match="Expected *"):
+        multitask_metrics.update(wrong_key_preds, _multitask_targets)
+    with pytest.raises(ValueError, match="Expected *"):
+        multitask_metrics.update(_multitask_preds, wrong_key_targets)
+    with pytest.raises(ValueError, match="Expected *"):
+        wrong_key_multitask_metrics.update(_multitask_preds, _multitask_targets)
```

Comment:
Could the matching message be more specific?

Reply:
Fixed